src/Data/Nat/Bounded.agda

```diff
@@ -0,0 +1,112 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Bounded natural numbers, and their equivalence to `Fin`
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Nat.Bounded where
+
+open import Data.Bool.Base using (T)
+open import Data.Nat.Base as ℕ
+import Data.Nat.DivMod as ℕ
+import Data.Nat.Properties as ℕ
+open import Data.Fin.Base as Fin using (Fin; zero; suc; toℕ)
+import Data.Fin.Properties as Fin
+open import Data.Product.Base using (_,_)
+open import Function.Base using (id; _∘_; _$_; _on_)
+open import Function.Bundles using (_⤖_; mk⤖)
+open import Function.Consequences.Propositional
+  using (inverseᵇ⇒bijective; strictlyInverseˡ⇒inverseˡ; strictlyInverseʳ⇒inverseʳ)
+open import Relation.Binary.Core using (_⇒_)
+open import Relation.Binary.Definitions
+open import Relation.Binary.PropositionalEquality using (_≡_; refl; cong; _≗_)
+open import Relation.Nullary.Decidable.Core using (recompute; map′)
+open import Relation.Nullary.Negation.Core using (¬_)
+
+private
+  variable
+    m n : ℕ
+
+------------------------------------------------------------------------
+-- Definition
+
+record ℕ< (n : ℕ) : Set where
+  constructor ⟦_⟧<_
+  field
+    value : ℕ
+    .bounded : value < n
+
+open ℕ< public using () renaming (value to ⟦_⟧)
+
+-- Constructors: '0 mod n', 'n-1 mod n', '1 mod n'
+
+⟦0⟧< ⟦-1⟧< : .{{ NonZero n }} → ℕ< n
+⟦0⟧<              = ⟦ 0 ⟧< >-nonZero⁻¹ _
+⟦-1⟧< {n = suc m} = ⟦ m ⟧< ℕ.n<1+n m
+
+⟦1⟧< : .{{ NonZero n }} → ℕ< n
+⟦1⟧< {n = 1}    = ⟦0⟧<
+⟦1⟧< {n = 2+ _} = ⟦ 1 ⟧< nonTrivial⇒n>1 _
+
+-- Projection from ℕ
```

Comment:
This is not really a 'projection' in the type-theoretic sense of the word, which at first confused me (I guess my brain's been infected in that way). Does anyone else use `π` for this? I'm really not liking this name.

Reply:
What do we call the canonical map from a thing to a quotient-thing? `q`?
We do usually call the canonical map from a sub-thing to a thing $$\iota$$, and $$\iota$$ for 'injection' and $$\pi$$ for 'projection' did make some sort of sense to me, at least...

... but for sure, not a 'projection' in the type-theoretic sense (a battle which ever rages in me between 'math' and 'type theory'... ;-))

Reply:
Much better.